### PR TITLE
Improve structure and documentation so docs.rs page looks nice

### PR DESCRIPTION
**Pull request by daniel-vainsencher**
# Intent:

Per the title. Specifically:
- [X] Link tutorial from top level doc
- [x] Show a (runnable?) example in the very first page.
- [x] Hide items that should not be API -- deferring for now.
- [x] Ensure all Structs are nicely named and briefly documented. (well, most)

Get more ideas [here](https://blog.guillaume-gomez.fr/articles/2020-03-12+Guide+on+how+to+write+documentation+for+a+Rust+crate)

# Validation:
Review documentation
- [x] Ensure examples still run.
- [x] If code of for_blog example changed, update blog :/
- [ ] ?
  
# State of PR
<Check boxes when done>

- [x] Rustdoc and implementation comments?
- [x] Self review comments on non-obvious changes?
- [x] It seems to you ready to merge on master?

### CODE REVIEW

File: examples/weighting_samples.rs [hunk 1]
```diff
@@ -4,11 +4,11 @@ use streaming_iterator::*;
 use utils::*;
 
 fn wd_iterable_counter_demo() {
-    println!("\n\n -----WdIterable Counter Demo----- \n\n");
+    println!("\n\n -----Weight Counter Demo----- \n\n");
```
Comment (daniel-vainsencher) [inline examples/weighting_samples.rs:7]: I did not rename some of the Yaml/File themed Iterables, but in principle we [should](https://rust-lang.github.io/api-guidelines/naming.html#iterator-type-names-match-the-methods-that-produce-them-c-iter-ty). Want to do them? can push such to this PR. 
I use `sed 's/BlaIterable/Blargh/g' -i */*.rs`.

File: src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+//! Implements [iterative
+//! methods](https://en.wikipedia.org/wiki/Iterative_method) and
+//! utilities for using and developing them as
+//! [StreamingIterators](https://crates.io/crates/streaming-iterator). A
+//! series of [blog
+//! posts](https://daniel-vainsencher.github.io/book/iterative_methods_part_1.html)
+//! provide a gentle introduction.
+//!
+//!
+//! ... but ok fine, here is a really quick example:
```
Comment (daniel-vainsencher) [inline src/lib.rs:12]: This is the documentation face of the crate, do you think it works?

File: src/lib.rs [hunk 3]
```diff
@@ -192,6 +251,8 @@ where
     timer: Instant,
 }
 
+/// Wrapper for Time.
+///
```
Comment (daniel-vainsencher) [inline src/lib.rs:255]: Leaving an empty line leads to the first to be used in the list on which the entry appears, and the rest to be used in the page for the entry.

File: Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "iterative_methods"
-version = "0.2.0"
+version = "0.2.1"
```
[the remark this answers is not in the document]
Comment (fox-daniel) [reply Cargo.toml:3]: roger that

**Pull request merged**